modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/MouseBasePhenotypeAnnotation.pm

```diff
@@ -0,0 +1,452 @@
+      $hash->{$key} = $value;
+      $column_headers->{$key} = 1;
+    }
+    $markers->{$hash->{marker_accession_id}} = 1;
+  }
+  close IN;
+
+  open(IN, "<".$coord_file) or die ("Could not open $coord_file for reading");
+  my $header;
+  my $marker_coords = {};
+  while (<IN>) {
+    chomp;
+    # test column names are still the same
+    if (/^1\./) {
+      my @values = split("\t", $_);
+      foreach my $value (@values) {
+        my ($column_number, $column_name) = split(/\s/, $value, 2);
+        $header->{$column_number} = $column_name;
+      }
+      unless ($header->{'1.'} eq 'MGI Marker Accession ID') { die 'Header in MGI_MRK_Coord.rpt has changed.'};
+      unless ($header->{'6.'} eq 'Chromosome') { die 'Header in MGI_MRK_Coord.rpt has changed.'};
+      unless ($header->{'7.'} eq 'Start Coordinate') { die 'Header in MGI_MRK_Coord.rpt has changed.'};
+      unless ($header->{'8.'} eq 'End Coordinate') { die 'Header in MGI_MRK_Coord.rpt has changed.'};
+      unless ($header->{'9.'} eq 'Strand') { die 'Header in MGI_MRK_Coord.rpt has changed.'};
+    }
+   my @values = split("\t", $_);
+    my $marker_acc = $values[0];
+    if ($markers->{$marker_acc}) {
+      unless ($marker_coords->{$marker_acc}) {
+        my $genes = $gene_adaptor->fetch_all_by_external_name($marker_acc);
+        if (scalar @$genes != 1) {
+          my $number_of_genes = scalar @$genes;
+          print $errFH "WARNING: Found $number_of_genes matching Ensembl genes for gene $marker_acc\n";
+          my $unique_seq_region_id = {};
+          foreach my $gene (@$genes) {
+            $unique_seq_region_id->{$gene->slice->get_seq_region_id} = 1;
+          }
+          print $errFH "WARNING: genes $number_of_genes are on different slices\n" unless (scalar keys %$unique_seq_region_id == 1);
+        }
+        next unless scalar @$genes;
+
+        my $marker_type = $values[1];
+        print $errFH "WARNING: type is not Gene (type = $marker_type)\n" unless ($marker_type eq 'Gene');
+
+        my $chromosome = $values[5];
+        my $start_coord = $values[6];
+        my $end_coord = $values[7];
+        my $strand = $values[8];
+
+        $marker_coords->{$marker_acc}->{type} = $marker_type;
+        $marker_coords->{$marker_acc}->{object_id} = join(";", map {$_->stable_id} @$genes);
+        $marker_coords->{$marker_acc}->{chromosome} = $chromosome;
+        $marker_coords->{$marker_acc}->{seq_region_id} = $genes->[0]->slice->get_seq_region_id;
+        $marker_coords->{$marker_acc}->{seq_region_start} = $start_coord;
+        $marker_coords->{$marker_acc}->{seq_region_end} = $end_coord;
+        $marker_coords->{$marker_acc}->{seq_region_strand} = ($strand eq '+') ? 1 : -1;
+      }
+    }
+  }
+
+  close IN;
+  return $marker_coords;
+}
+
+sub parse_mouse_phenotype_data {
+  my ($self, $infile, $marker_coords, $data_source, $source_name2ids, $variaiton_dba) = @_;
+
+  my $individual_adaptor = $variaiton_dba->get_IndividualAdaptor;
```

Comment:
Mispelled `$variaiton_dba`
Possibly could be used as object variable (got also `$core_dba`, `$variation_db` as method parameters)

Reply:
method now renamed to parse_input_file